experimental/uefi/app/src/main.rs

```diff
@@ -70,53 +67,31 @@ fn main(handle: Handle, system_table: &mut SystemTable<Boot>) -> Status {
     serial_echo(handle, system_table.boot_services()).unwrap();
 }
 
-fn echo_loop(serial: &mut Serial) -> Result<!, uefi::Error<()>> {
-    let mut buf: [u8; 1024] = [0; 1024];
-
+fn serial_echo(handle: Handle, bt: &BootServices) -> Result<!, uefi::Error<()>> {
```

Comment:
Could you please add comments about what the arguments are.
What are boot services? And is handle just a serial port number or something else?

Reply:
I've added a bit of documentation, but unfortunately it's a bit generic (boot services are, uh, the UEFI Boot Services, as defined by the UEFI specification).